### DISCUSSION TITLE
Unsure about handling error responses with useQuery

### DISCUSSION THREAD

**Discussion by fdjones** [TanStack/query#1099] [General] [answered]
Hi, I got stuck recently when working with useQuery and trying to handle error responses from Axios. I think I understand what the issue is, but I'm not sure that I fully grok the correct pattern to implement instead. Here was the issue I was having:

```
// signout.js 

function signout() {
  return axios
    .get('/api/auth/logout')
    .catch((err) => { console.log('err:'); throw err; }); // on a 440, this catches (as you would expect, I only added this here for debugging purposes)
}

export default function useSignOut() {
  const { refetch } = useQuery('signout', signout, {
    enabled: false,
    retry: false,
  });

  return refetch;
}
```

that was being used in a components click handler, like so:

```
// Logout.js

  function handleSignOut() {
    signOut()
      .then(persistor.purge) // all the `.then`'s are hit?? shouldn't it be a .catch?
      .then(() => dispatch({ type: 'signout' }))
      .then(() => navigate('/signin'));
  }
```

So, I think I understand what's happening. react-query doesn't throw as long as the `signout` function returns a response, and the correct way to handle errors in the component would be to do something like `const { data, isError}` and then check the value of `isError`. However, this feels clunky when you want to do something as the result of a click, where it would be easier to just handle any error response in the promise chain.

I know that useMutation has the config option `throwOnError` which feels like the same thing that I'm wondering about here with useQuery? I'm sure there's a reason why it's provided for useMutation and not useQuery, I'm just wondering what that is :)

So yeah, this isn't a feature request but more of a "what am I missing here and what's the best way of doing this?". 

Thanks

**Comment by tannerlinsley** [comment, marked as answer]
useQuery should Never really be used for mutations to the server. Sign out is definitely a mutation. By using useQuery, you’re potentially and probably running the sign out function many times more than expected.

**Reply by fdjones** [reply]
@tannerlinsley Thanks for the response - I didn't initially consider it a mutation but thinking about it I suppose I'm mutating the sessions table so it would make more sense to use `useMutation` here. Cheers.
